### DISCUSSION TITLE
Mapping WHILE cycling

### DISCUSSION THREAD

**Discussion by canfe** [streetcomplete/StreetComplete#4395] [Ideas]
It would be helpful if as you ride, the app would automatically propose questions when you get to the point.
So that you could STOP and answer them.  ;-)
Currently you have to tap on the screen in order to see the question and it is not safe while cycling.

**Comment by FloEdelmann** [comment]
You mean like playing a sound / triggering a notification when near a quest while the screen is off?

See #48 for that suggestion and why it was ultimately rejected.

**Comment by canfe** [comment]
As a cyclist I would essentially use it to define the quality of bike routes, so I would prefer the screen to come on since I keep my cell phone on my handlebars.

For walkers, of course, a sound would be better since they keep it in their pocket.

I thank you so much for your interest. Do you think the modification will be considered? We are several now interested in it.

**Reply by mnalis** [reply]
See the comment https://github.com/streetcomplete/StreetComplete/issues/48#issuecomment-908380719 with exact reasons why it was rejected. Did you think any of them change significantly, or could you suggest more effective solutions to some of them? If not and the situation remains the same, it is likely the rejection reasons would remain the same too. (There were several people interested back then too, myself included, but the barriers to implement it were too too many and too high). 

If however there are dramatic new insights on the problems, everything is possible...

**Reply by Helium314** [reply]
I was considering to give it a try, but IMO the biggest issue here is the background download.
Downloading and persisting data takes a lot of processing time, and in my experience Android seems to strongly limit resources when the screen is off. It's very noticeable when uploading a lot of edits, I found that often only a few edits are uploaded in several minutes when the screen is off (much slower than with screen on).
So it might easily happen that persisting is finished only after the user already moved out of the downloaded area.

Additionally, it also puts (somewhat) unnecessary load on OSM servers, but I really can't tell how much of an issue that is.

**Comment by canfe** [comment]
@mnalis I think 'times have changed' and that StreetComplete is becoming a killer application and therefore needs to take some risks to stay on top.
Concretely: a cyclist who wants to 'map without putting his feet on the ground' can only look at a few things, essentially the road where he pedals and little else.
Wouldn't the various problems listed be alleviated acceptably if the search area were reduced to just a few metres?

**Reply by mnalis** [reply]
If trimmed bbox down to few meters, I think OSM probably wouldn't download anything (you really want whole way loaded, or at least two points, which might be hundreds or more meters away if straight piece of the road)... Do you perhaps plan bicycle trips in advance? I often do, and use [OsmAnd](https://en.wikipedia.org/wiki/OsmAnd) to navigate me. It can take custom GPX track with waypoints (and audio warn / turn on the screenwhen you reach them). 

Perhaps a way around the problem it to use overpass (like [this one](http://overpass-turbo.eu/s/1m2O) for non-primary non-motorway roads without `smoothness=*` or `surface=*`) to select interesting roads on the way, and export the GPX, and use that to warn you when you reach that interesting places.

However, my experience when enabling notification of interesting problems in [Vespucci](https://wiki.openstreetmap.org/wiki/Vespucci) was that there were **_way too much_** notifications. I (at least) certainly am not going to get down from bicycle every 50-100 meters for any significant amount of time... At least, you can use _**overpass example above to see how spammy**_ just those two quests (surface & smoothness) would be in areas you're traveling in.

Another alternative that I've done before is to setup your OsmAnd to record video (this will also eat some battery, and mobile phone space), and then later fire up JOSM to add surface/smoothness/lit to ways I passed that did not have information. (there are smarter tools to do this smarter too, but they're nowdays owned by evil companies so I'm going to advertise them). 

Or, stop on (almost) every crossing and open a note detailing which surface/smoothness/lit this road had. (this _is_ tiring, but I still use variant of that, noting mostly only when things change or enough time has passed, and than later correlate recorded GPX track with those notes. But I had to fire up JOSM quickly thereafter, or I soon forget exactly what I meant in some notes)

**Reply by mnalis** [reply]
As an example, the overpass query in relatively unmapped (regarding surface/smoothness) about 1km driving, found about 10 quests withing radius of 100m of the road I' plan to take. That is having 1 alert about **_every 100m, when I'd have to stop the bicycle_** and solve the quest. To me personally, that would be way **_too often_**, even if all the technical hurdles could be overcome. (The main problem is that the roads are often split into many parts, especially if there are any crossings, or speed changes, or villages etc, and each part or the road is its own OSM way with its own quest.)

For me as cyclist, for such SC feature to work, it would need to notify me for quest **at most** once every 1 km (ant that only if my primary purpose when cycling was to go out and map so I'm willing to stop every 2-3 minutes). And if my primary purpose was not mapping when I'm cycling, I'd be surprised if I'd tolerate alert to stop and get off the pedals more often than once every 15-20 minutes (i.e. at most 1 quest every 5-6km)

What would be your tolerances, @canfe (i.e. how often are you prepared to stop and get off the pedals)? And do you care about any any other quests other than `surface` and `smoothness` when cycling?

**Comment by canfe** [comment]
@Helium314 Yes it is also true in my experience.

This is another thing to put into 'cyclist mode' ;-)

**Reply by mnalis** [reply]
This is an Android (or any modern phone issue) instead of StreetComplete issue, see for example https://dontkillmyapp.com/

Most modern phones are bent on having "best battery runtime benchmarks", and the way they do it is by sleeping majority of the time, **regardless** (or in spite of!) of the user or the app actually **want**. And it is only getting worse with each new version of the phone. So basically only realistic way to fix it is to use some very old smart phone with old version of Android (or perhaps root the newer phone and  downgrade to Android versions without battery saving _"helping"_ features).

So I don't think it would work well (if at all). Maybe if you preloaded all the data before trip (if you know where you were going to go) e.g. https://github.com/streetcomplete/StreetComplete/issues/3256 it might work somewhat better, but still no guarantees.

**Comment by canfe** [comment]
I would never really want to stop.

Slow down, but don't stop. (by the way I go quite slowly on the bike, I observe more than pedal)

That's why if you could reduce the radius to 10 metres it would be more than enough to upgrade the quality of the road surface.

**Reply by mnalis** [reply]
Oh. I did take that _"So that you could STOP and answer them"_ at face value, and it seems to have been an sarcasm. 

Driving and using mobile phone is not legal over here, and sounds unsafe; but if it _is_ legal and safe in your case, then you'd also probably be just fine with keeping StreetComplete in foreground, and enabling `Keep screen on` in `Settings`. If increased power usage is the main/only problem there, that issue could probably much easier be solved simply be using cheap external battery bank.

Another possible alternative to StreetComplete I've stumbled upon (note: I don't know if it actually works) is [What's Nearby](https://github.com/Teester/Whats-Nearby/) which seems to have been created in response to exactly this [inability of StreetComplete](https://github.com/Teester/Whats-Nearby/issues/6#issuecomment-366572726)

**Reply by mnalis** [reply]
Or @canfe, you may want to try EE fork which has implemented notifications for nearby quests (see https://github.com/streetcomplete/StreetComplete/issues/48#issuecomment-1279382654)
